src/main/java/secondmind/TaskList.java

```diff
@@ -0,0 +1,117 @@
+        if (taskInfo.length == 1) {
+            throw new EmptyToDoException();
+        }
+        taskInfo[0] = "";
+        String taskDescription = String.join(" ", taskInfo);
+        taskDescription = taskDescription.substring(1);
+        return new ToDoTask(taskDescription);
+    }
+
+    private Task createDeadline(String task) {
+        String[] taskInfo = task.split(" /by ");
+        //Prefix of taskInfo[0] is "deadline "
+        String taskDescription = taskInfo[0].substring(9);
+        String taskDeadline = formatDateTime(taskInfo[1]);
+        return new DeadlineTask(taskDescription, taskDeadline);
+    }
+
+    private Task createEvent(String task) {
+        String[] taskInfo = task.split(" /", 3);
+        //Prefix of taskInfo[0] is "event "
+        String taskDescription = taskInfo[0].substring(6);
+        //Prefix of taskInfo[1] is "from "
+        String taskStart = formatDateTime(taskInfo[1].substring(5));
+        //Prefix of taskInfo[2] is "to "
+        String taskEnd = formatDateTime(taskInfo[2].substring(3));
+        return new EventTask(taskDescription, taskStart, taskEnd);
+    }
+
+    private Task createTask(String task) 
+            throws EmptyCommandException, EmptyToDoException, UnknownCommandException {
+        String[] taskInfo = task.split(" ");
+        String taskType = taskInfo[0];
+        if (taskInfo[0].equals("")) {
+            throw new EmptyCommandException();
+        } else if (taskType.equals("todo")) {
+            try {
+                return createToDo(taskInfo);
+            } catch (EmptyToDoException e) {
+                throw e;
+            }
+        } else if (taskType.equals("deadline")) {
+            return createDeadline(task);
+        } else if (taskType.equals("event")) {
+            return createEvent(task);
+        } else {
+            throw new UnknownCommandException();
+        }
+    }
+
+    public Task addToTaskList(String task) 
+            throws EmptyCommandException, EmptyToDoException,  UnknownCommandException, DateTimeParseException  {
+        Task curr = createTask(task);
+        taskList.add(curr);
+        return curr;
+    }
+
+    public ArrayList<Task> getTaskList() {
+        return this.taskList;
+    }
+
+    public void markAsDone(int taskNumber) throws InvalidTaskNumberException {
+        if (taskNumber <= 0 || taskNumber > getTaskCount()) {
+            throw new InvalidTaskNumberException(taskNumber);
+        } else {
+            Task curr = this.taskList.get(taskNumber-1);
+            curr.markAsDone();
+        }
+    }
+
+    public void markAsUndone(int taskNumber) throws InvalidTaskNumberException {
+        if (taskNumber <= 0 || taskNumber > getTaskCount()) {
+            throw new InvalidTaskNumberException(taskNumber);
+        } else {
+            Task curr = this.taskList.get(taskNumber-1);
+            curr.markAsUndone();
+        }
+    }
+
+    public void delete(int taskNumber) throws InvalidTaskNumberException {
```

Comment:
1. I love that you named all the methods appropriately, it leaves very little room for ambiguity. 😄 
2. Might `delete(int)` be confused for deleting the entire list instead of a task? (might be clearer to use `deleteTask`)